### ISSUE TITLE
[BUG] Connection to rouvy unstable

### ISSUE THREAD

**Issue by caligrom** [cagnulein/qdomyos-zwift#2910]
**Describe the bug**
Unstable connection to rouvy and clunky way of getting the connection. App gets sometimes very slow - can only be fixed by disconnecting the bike from power and restarting the phone

**To Reproduce**
1. Connect bike to QZ (on Android, Samsung) - works
2. Start rouvy on Windows PC
3. Search for "JUSTO" named QZ phone
4. Find several JUSTO entries, try to select the right one
5. Wait until connection is established
6. After connected, start ride
7. Sometimes App gets very slow
8. Sometimes connection is interrupted after minutes

**Expected behavior**
Easy and stable connection. Preferably a way to regulate riding by power with automatically adjusted resistance

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - Rouvy - Windows 10

**Smartphone (please complete the following information):**
- Samsung A32 - QZ 2.16.66 directly from google play

**Append a debug log**
https://drive.google.com/file/d/15VchkEY33l4mDQkhoqqWC9MaMRM5DRY3/view?usp=drive_link
https://drive.google.com/file/d/16-A83CrMBIQS3M5bZiXJJRQHUJ1IDPS5/view?usp=drive_link

  
  
**Additional context**

**Comment by cagnulein** [comment]
Hi, unfortunately from a fit file I can't understand your issue. Please
create a debug log
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

When you are saying that it is slow, the metrics are updating? If not I
guess the phone has not enough power, or you have power saving enabled, or
you have some apps on background that are eating cpu

Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 16 dic 2024 alle 20:48 caligrom ***@***.***> ha
scritto:

**Comment by caligrom** [comment]
Ok, sorry - wrong file.

Here are the correct debug files:
[debug-Sun_Dec_15_15_34_49_2024.log](https://github.com/user-attachments/files/18156839/debug-Sun_Dec_15_15_34_49_2024.log)
[debug-Sun_Dec_15_15_36_48_2024.log](https://github.com/user-attachments/files/18156843/debug-Sun_Dec_15_15_36_48_2024.log)

Phone is normally running smoothly - with slow I mean the metrics are slow but also if I perform an input (e.g. in the settings menu or shifting gears) its very laggy.
The phone is my work phone where a second "work" profile is running in parallel - perhaps that is part of the issue? No other apps are running and the phone is not in energy safe mode / low power.

Best wishes,
Stefan

**Comment by cagnulein** [comment]
Ok, I'll take a look. But it sounds that the phone is not powerful enough
to run all of this.
For example the time metrics is jumping some seconds?


Il giorno lun 16 dic 2024 alle 21:45 caligrom ***@***.***> ha
scritto:

**Comment by caligrom** [comment]
Yes, sometimes this is happening - time metric is jumping or also wattage / cadence are lagging - however sometimes it also runs smoothly.

**Comment by cagnulein** [comment]
@caligrom i tried to download your logs but I can't, do you mind upload them again?

**Comment by caligrom** [comment]
[debug-Sun_Dec_15_15_34_49_2024.log](https://github.com/user-attachments/files/18188950/debug-Sun_Dec_15_15_34_49_2024.log)
[debug-Sun_Dec_15_15_36_48_2024.log](https://github.com/user-attachments/files/18188955/debug-Sun_Dec_15_15_36_48_2024.log)

@cagnulein: Here are the files

**Comment by cagnulein** [comment]
@caligrom same thing. i can't download them. are you able to? maybe try to zip them?

**Comment by cagnulein** [comment]
@caligrom Any news?

**Comment by JLou5641** [comment]
Hello,

I "re-open" this issue. I just discover this app, perfect for virtual gearing.
I have exactly the same issue with Rouvy.

It work perfectly with zwift for connection and riding, but not with Rouvy.
It hardly connect, unstable, and i need to disconnect and reconnect QZ from Rouvy each time i start Rouvy.
Yesterday again, Rouvy loose the conncection with qz ( even if it was still connected in Rouvy setting ), while QZ keep connection and statistic from my kickr Core.

I tested it with my Redmi Note 13 or Oukitel RT3 Pro, with last rouvy update ( wich add support for zwift cog and clic ), and windows 11 pro on a recent gamer laptop.

Greetings.

**Comment by cagnulein** [comment]
Qz on android? Did you rename to JUSTO  ?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 16 feb 2025 alle 10:17 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Lol, you answer faster than i am on my bike ^^.

Nope, I didn't rename anything, apart my android device with a short bluetooth name

**Comment by cagnulein** [comment]
Haha, Ok give it a try and let me know


Il giorno dom 16 feb 2025 alle 10:57 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
You mean, i need to rename my android bluetooth to JUSTO?

**Comment by cagnulein** [comment]
Yep

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 16 feb 2025 alle 11:10 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
okey, let me some minutes

**Comment by JLou5641** [comment]
Nope; it change, nothing. This time Rouvy loose the connectin after 1min30 unfortunaly

**Comment by cagnulein** [comment]
But on zwift are you using the Bluetooth name from qz or the wahoo kickr
virtual device from qz?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 16 feb 2025 alle 11:23 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Let me try zwift... 

About Rouvy, i restart it, this time it reconnect automaticaly to QZ.

**Comment by cagnulein** [comment]
Consider that rouvy it behaves in a different way from zwift. Every time
you have to pair it again because they are using the mac address and the
mac address of the phone is changing every time

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 16 feb 2025 alle 11:29 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Okey, noted for Rouvy Mac Adress.

So, for Zwift, i made little try with Bluetooth name from QZ, no problem. It worked perfectly

**Comment by cagnulein** [comment]
That's so strange, if you want I can check with a debug log from Qz if
there is a difference on your setup from a zwift and rouvy to compare them
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 16 feb 2025 alle 11:44 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Okey, i will take a look later. I need to go ;)

**Comment by JLou5641** [comment]
Hello, i just made a ride on rouvy. This weekend i reset everything... 
Rouvy won't work with QZ... Few time after beginning, Rouvy loose the QZ connection.
I have no issue with Zwift and MyWoosh

**Comment by cagnulein** [comment]
So I guess rouvy is buggy on your device. You can try to swap them to see if it makes any difference

**Comment by JLou5641** [comment]
I use rouvy on Windows 11.
The result is same with my tablet or smartphone.. It's like i'm "ban" of
the ride after few minutes.

I will continue to test and investigate.

Le lun. 24 févr. 2025, 19:54, Roberto Viola ***@***.***> a
écrit :

**Comment by cagnulein** [comment]
I mean if qz on the phone, did you try qz on the  tablet and rouvy on the
phone?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 24 feb 2025 alle 19:57 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
aah ! No, i didn't do this.
I will do and make feedback

Le lun. 24 févr. 2025, 20:37, Roberto Viola ***@***.***> a
écrit :

**Comment by JLou5641** [comment]
Question,
Is the connection between QZ and Laptop with Windows through wifi or bluetooth ?

**Comment by cagnulein** [comment]
because otherwise rouvy doesn't identify your phone as a compatible trainer. in this way rouvy will see the phone as a trainer instead

**Comment by JLou5641** [comment]
okey, thanks for all this information..
I will have some freetime this late afternoon, I will make some different
test over bluetooth:  Qz + Rouvy Android, QZ + Rouvy over Android Emulator
on Windows, QZ + another extra bluetooth key ( if i find it on my stuff 😅
).
I would like to make QZ work perfectly instead of buying Cog + Click.

Le mar. 25 févr. 2025, 09:25, Roberto Viola ***@***.***> a
écrit :

**Comment by cagnulein** [comment]
bluetooth on android emulator doesn't work

**Comment by JLou5641** [comment]
okey👍

Le mar. 25 févr. 2025, 09:33, Roberto Viola ***@***.***> a
écrit :

**Comment by JLou5641** [comment]
I made some test.

With Rouvy on Android, it seem working perfectly.
Strange thing appear on Rouvy Windows, i made 15min of training session with no issue. It's the first time i tested it on a training session and worked perfectly more than 5min... But as soon as I exit the training session to make a simple ride, i had "no connection" directly at the start of the ride. Then i just exit the ride and relaunch it directly, without disconnecting and reconnecting QZ virtual bike from the home trainer connection screen on rouvy, so just exit the ride and re-launch it, then connection worked..... but after 3min, i loosed it... It's like a "BAN", not a real deconnection
I don't know what to think, if it's an issue on my Laptop or something on Rouvy PC wich detect QZ,
But, I remind you that Zwift work perfectly both with Virtual Device or Justo/Bike bluetooh...

I join the debug log if you want to check



Greetings,

[debug-Tue_Feb_25_18_02_48_2025.log](https://github.com/user-attachments/files/18970091/debug-Tue_Feb_25_18_02_48_2025.log)

**Comment by cagnulein** [comment]
Could be the Bluetooth dongle on the pc?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 25 feb 2025 alle 18:59 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
I used the internal laptop bluetooth chip. I didn't find my usb dongle i had by the past.

**Comment by JLou5641** [comment]
zwift work perfecty with kicker virtual device from QZ or with the bluetooth "QZ" name. I never test more than 15min cause i don't have membership on it, but i never had "deconnection" on zwift while it happen only after few minutes on Rouvy windows.

**Comment by cagnulein** [comment]
On rouvy logs there is something?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 25 feb 2025 alle 19:37 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Where can i find rouvy log?

**Comment by cagnulein** [comment]
Rouvy log files are stored in the following locations:

• Windows: C:\Users\[YourUsername]\AppData\LocalLow\VirtualTraining\ROUVY\

• macOS: /Users/[YourUsername]/Library/Application
Support/eu.virtualtraining.rouvy.mac/


Replace [YourUsername] with your actual username. These directories contain
data related to your Rouvy activities and settings. Please note that
modifying or deleting files in these folders can affect the application’s
performance.


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 25 feb 2025 alle 19:40 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Yes, there is an error.. I just copy and past the log here when it loose "connection" after 3min17:

025/02/25 18:18:40.448  INFO Rouvy.Backend.AR.Avatar.AssetInstanceManagerUnloadHeuristics - Unloading unused garage assets...
2025/02/25 18:18:40.667  INFO Rouvy.Backend.Logging.BffClientLogger - Get activity media urls operation finished [Urls: 1]
2025/02/25 18:18:41.815  INFO Rouvy.Backend.Logging.BffClientLogger - Home featured swimlane request completed [Records: 3]
2025/02/25 18:18:42.035  INFO Rouvy.Backend.Logging.BffClientLogger - Get challenges returns 15/15
2025/02/25 18:18:42.304  WARN Rouvy.Backend.Device.Ble.Service.BleService - [{ Device = { Name = , Address = 4995217DE3A1 }, Service = 00001826-0000-1000-8000-00805f9b34fb, Characteristic = 00002ad9-0000-1000-8000-00805f9b34fb }]Characteristic write failed: BleHWPluginError:
Error: 0x5101F
Framework: Bluetooth Framework Error Codes
Category: Bluetooth Low Energy error codes
Constant: WCL_E_BLUETOOTH_LE_COMMUNICATION_FAILED
Description: The target server did not provide an appropriate network response.
-Description: Error when reading value
2025/02/25 18:18:42.342  INFO Rouvy.Backend.Logging.BffClientLogger - Operation returns route detail for RouteId: "93695"
2025/02/25 18:18:42.428  INFO Rouvy.Backend.Device.Ble.HwInterface.Win.WclBleGattThread - GATT disconnected: WinBlePeripheralInfo: [4995217DE3A1], reason: "Error: 0x30008
Framework: Connections Error Codes
Category: Common connection errors
Constant: WCL_E_CONNECTION_CLOSED
Description: The connection has been closed."
2025/02/25 18:18:42.428  INFO Rouvy.Backend.Device.Ble.HwInterface.Win.BleWinHwInterface - Device disconnected WinBlePeripheralInfo: [4995217DE3A1] - Success, BleHWPluginError:
Error: 0x30008
Framework: Connections Error Codes
Category: Common connection errors
Constant: WCL_E_CONNECTION_CLOSED
Description: The connection has been closed.
-Description: GATT disconnect reason Ox30008
2025/02/25 18:18:42.818  INFO Rouvy.Backend.Logging.BffClientLogger - Operation returns 2 route collections
2025/02/25 18:18:43.103  INFO Rouvy.Backend.Device.Ble.HwInterface.Win.WclBleGattThread - GATT connection connected - WinBlePeripheralInfo: [4995217DE3A1] , reason - 0
2025/02/25 18:18:43.117  INFO Rouvy.Backend.Device.Ble.HwInterface.Win.BleWinHwInterface - Device connected WinBlePeripheralInfo: [4995217DE3A1]
2025/02/25 18:18:43.257  INFO Rouvy.Backend.Logging.BffClientLogger - Latest notifications list operation finished [Records: 16]
2025/02/25 18:18:43.350  INFO Rouvy.Backend.Logging.BffClientLogger - Persist activity media operation finished [ActivityId: "250225171840-c7c590ff"]
2025/02/25 18:18:43.385  INFO Rouvy.Backend.Logging.BffClientLogger - Spotlight data retrieved successfully.
2025/02/25 18:18:43.643  INFO Rouvy.Backend.Logging.BffClientLogger - Spotlight swimlane data retrieved successfully.
2025/02/25 18:18:43.824  INFO Rouvy.Backend.Logging.BffClientLogger - Spotlight data retrieved successfully.
2025/02/25 18:18:44.184  WARN Rouvy.Backend.Device.Ble.Service.BleService - [{ Device = { Name = , Address = 4995217DE3A1 }, Service = 00001826-0000-1000-8000-00805f9b34fb, Characteristic =  }] Service failed
2025/02/25 18:18:44.184  WARN Rouvy.Backend.Device.Ble.Service.BleService - [{ Device = { Name = , Address = 4995217DE3A1 }, Service = 00001826-0000-1000-8000-00805f9b34fb, Characteristic = 00002ad2-0000-1000-8000-00805f9b34fb }] Subscription failed: BleHWPluginError:
Error: 0x50015
Framework: Bluetooth Framework Error Codes
Category: Classic Bluetooth error codes
Constant: WCL_E_BLUETOOTH_UNEXPECTED
Description: Unexpected or unknown Bluetooth error.
-Description: Error when subscribing Characteristics
2025/02/25 18:18:44.201  WARN Rouvy.Backend.Device.Ble.Service.BleService - [{ Device = { Name = , Address = 4995217DE3A1 }, Service = 00001826-0000-1000-8000-00805f9b34fb, Characteristic = 00002ada-0000-1000-8000-00805f9b34fb }] Subscription failed: BleHWPluginError:
Error: 0x50015
Framework: Bluetooth Framework Error Codes
Category: Classic Bluetooth error codes
Constant: WCL_E_BLUETOOTH_UNEXPECTED
Description: Unexpected or unknown Bluetooth error.
-Description: Error when subscribing Characteristics
2025/02/25 18:18:44.217  WARN Rouvy.Backend.Device.Ble.Service.BleService - [{ Device = { Name = , Address = 4995217DE3A1 }, Service = 0000180d-0000-1000-8000-00805f9b34fb, Characteristic = 00002a37-0000-1000-8000-00805f9b34fb }] Subscription failed: BleHWPluginError:
Error: 0x50015
Framework: Bluetooth Framework Error Codes
Category: Classic Bluetooth error codes
Constant: WCL_E_BLUETOOTH_UNEXPECTED
Description: Unexpected or unknown Bluetooth error.
-Description: Error when subscribing Characteristics
2025/02/25 18:18:44.217  WARN Rouvy.Backend.Device.Ble.Service.BleService - [{ Device = { Name = , Address = 4995217DE3A1 }, Service = 0000180d-0000-1000-8000-00805f9b34fb, Characteristic =  }] Service failed
2025/02/25 18:21:08.068  INFO Rouvy.Backend.AppBootstrapper - AppBootstrapper.Quitting

**Comment by cagnulein** [comment]
It doesn't seem a ban at all. This seems a Bluetooth incompatibility. I
would give a try to a different Bluetooth dongle like the bt400

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 25 feb 2025 alle 20:38 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
yes. I'm agree... It remind me a "bug" i had by the past with PS3 gamepad and the laptop bluetooth... It worked only well with an external dongle through the custom gamepad app... Cause PS3 Gamepad not working directly, like you made with QZ for some Home Trainer...

I presume the BT500 will work to, or do i need to prefer the BT400? ( The BT500 is cheaper )

**Comment by cagnulein** [comment]
Bt400 is way better if you can

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 25 feb 2025 alle 20:48 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Okey, I buy it. But can't make test before next Monday as i need to move when i will receive the dongle.

**Comment by cagnulein** [comment]
Ok, no problem.


Il giorno mar 25 feb 2025 alle 20:51 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Hello,

I did a ride Yesterday with the BT400.
I confirmed that the laptop embedded bluetooth chip were disabled.
So, after 10min of riding, I though it worked, but near 15th minutes, it did the same "disconnection" 😩.
It just took more minute than previous test last week with the embedded intel bluetooth..

I really don't know what happened. I made all the possible test.. With different android device for QZ, different Bluetooth name, bluetooth dongle, reset everything..
There is no reason that bluetooth "loose connection".. And it's not real deconnection. It just loose data.

I would like to know if someone have same config like me.

**Comment by cagnulein** [comment]
yes they are plenty of rouvy users, you can check here and also on the qz facebook group.
Are you able to connect QZ only as resistance on rouvy and leave the power and cadence on the trainer directly @JLou5641  ?

**Comment by JLou5641** [comment]
I can connect all from qz to rouvy. But i need to connect the resistance first, and then power and cadence. 
Can i use qz only for resistance, and legit Kick for power and Cadence?

**Comment by cagnulein** [comment]
yes if the kickr allows to have double bluetooth connection. one for qz and
one for rouvy. most of the trainers can

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 4 mar 2025 alle ore 09:38 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
okey, good news, yes it's possible.
And maybe, this "double connection" is what make the trouble....
I will give it a try like like this.

Thanks for your great disponibility for support 👍

**Comment by cagnulein** [comment]
I need a debug log to check this

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 4 mar 2025 alle 19:31 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
I will send it to you tomorrow

**Comment by JLou5641** [comment]
Hi,

here the log file:

https://drive.google.com/file/d/1KmbO0swdVA8PRrRGJ1mDqisLPBh7cn1p/view?usp=drivesdk

**Comment by cagnulein** [comment]
Thanks I will check it ASAP

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 4 mar 2025 alle 22:44 JLouLoulou ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
Access requested to the file. thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 5 mar 2025 alle ore 05:13 cagnulein ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
unfortunately it's a 0 byte file. could you please check it? you can zip it
and put it also here

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 5 mar 2025 alle ore 06:18 JLouLoulou ***@***.***>
ha scritto:

**Comment by JLou5641** [comment]
Here it is

[debug-Tue_Mar_4_18_25_04_2025.zip](https://github.com/user-attachments/files/19083557/debug-Tue_Mar_4_18_25_04_2025.zip)

**Comment by cagnulein** [comment]
i checked: QZ is sending correctly inclination and gears both: so maybe rouvy is overriding it somehow? just to understand, in the same scenario, the resistance is working fine on zwift for example? also if you instead pair everything on qz to rouvy, the resistance is fine?

**Comment by JLou5641** [comment]
yes, with only resistance connected to QZ, and power and cadence directly to kickr, i suspected that Rouvy start to override QZ at the beginning of the ride. Because, the firsts km, feeling were good, changing gear according to slope were good, but then, after around five minutes i think, slope didn't affect résistance anymore, so i stayed on the same gear ( 2 same Gears in fact ) all the time no matter the slope.
You can see it on the recap graphics on my Rouvy Ride profil, at the begining you can see the power chaging depend the slope, and then the power were regular all the time until the end of the ride.
I will make a test with previous version of rouvy like you suggest on Facebook group, i wil download and install the version just before zwift Cog and Click update compatibility + virtual gearing.

Yes for Zwift + QZ everything work fine with no disconnection, and in Rouvy + QZ for all sensor, everything work well until it disconnect after few minutes. 
So, recap is:
Rouvy + full QZ sensor : Everything work well few minutes before "Rouvy disconnection"
Rouvy + Resistance QZ only: Everything work well few minutes before resistence stay constant like ERG mode, but no disconnection.

I really suspect something from Rouvy with the kickr, maybe unintentionnaly from developer, maybe a "bug"...
There is no reason that other app worked well but not rouvy with QZ and wahoo Kickr zwift cog compatible

I asked on Facebook for someone who have same config than me... No result for now.

**Comment by JLou5641** [comment]
yes for sure.
And i will test also with Windows version of Qz, i still didn't try it.. maybe it could have different behavior.... who knows?😅

**Comment by JLou5641** [comment]
Hi,

I made a try with Rouvy 3.9.5, wich is the version the before Zwift Clic and Cog integration... It change nothing..

Rouvy "disconnect" from QZ even if it's still connected, and i need to exit and restart the ride to make it working.

I take you 2 screenshot to show what how it is when it disconnect

![Image](https://github.com/user-attachments/assets/f537a74b-bfae-4fd8-8211-570f357dc497)

![Image](https://github.com/user-attachments/assets/6eccf96c-98e8-4bb8-96f3-cb93c7c3ef70)

Just go to option and go back to ride don't work.. I need to entirely exit the ride or fully disconnect and reconnect in option during the ride.

I really don't know what to do apart buying a Zwift Click for virtual gearing :/ ... There is really something between Rouvy Windows and kickr wich not allow working more than some minutes.. I have a Dektop PC that i need to reassemble with wich i could test, but i have a doubt that it will change something. If it was the bluetooh, the problem should be solved cause i use a BT400 now.

**Comment by JLou5641** [comment]
Woh woh woh!!!

I find how to make it work! Totaly by chance!
I proved that rouvy not working like other app or in android with kickr ( and probably all few HT which worked with clic and cog )... 
I'm sure you will not expect that.

No need to rename android bluetooth, cause, no need to connect Rouvy to QZ!!! 
I would continue my ride before going to my handball training.. so i update again rouvy to last version, restart my kickr.. Restart my ride.... Then i took my tablett just for fun... and qz... and this is at this moment i see QZ connecting to my kickr, read all rouvy data value, and change my kick resistence! So gearing was working!
Everyting worked well during all the session. I take you a screenshot with my phone to show you that Rouvy is connected directly to my Kickr, and QZ to my kickr too.. Amazing! 

![Image](https://github.com/user-attachments/assets/1241ddb6-48f9-4fcb-bbe2-2d80c1d368ad)

I need to go.. Will discuss about it later.. But i have an idea how kickr work with rouvy and virtual gearing

**Comment by cagnulein** [comment]
Hah Ok so yes rouvy is controlling the inclination on the trainer but qz is sending gears right? That's awesome!

**Comment by JLou5641** [comment]
yes, it's that, Rouvy send only inclination to Kickr, the the kicker send his info again to Rouvy, but in same time work with QZ ... I think QZ really work like the Zwift Clic, because virtual gear are programmed in the firmaware, and QZ send gear configuration to the kickr, like you should do if you have the real Zwift Clic were connected to rouvy. So it means in virtual gear with clic, Rouvy send data for virtual gear ( which are not customisable for now but it will should in the future, i talked to them about that )to the kickr like if an inddpendant application and then the kickr send the result for power to rouvy... So QZ work exactly same. It connect direcly to kickr to send the gear data.

I hope i explained well

**Comment by cagnulein** [comment]
That's great it's a win win so!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 5 mar 2025 alle 22:06 JLouLoulou ***@***.***>
ha scritto:
